tests/test_odilo.py

```diff
@@ -0,0 +1,342 @@
+        self.api.queue_response(200, content=checkin_json)
+
+        response = self.api.checkin(self.PATRON, self.PIN, self.licensepool)
+        eq_(response.status_code, 200,
+            msg="Response code != 200, cannot perform checkin for record: " + self.licensepool.identifier.identifier
+                + " patron: " + self.PATRON)
+
+        checkout_returned = response.json()
+        print 'Checkout returned: %s' % checkout_returned['id']
+
+    #################
+    # Patron Activity tests
+    #################
+
+    def test_61_return_hold_patron_not_found(self):
+        patron_not_found_data, patron_not_found_json = self.sample_json("error_patron_not_found.json")
+        self.api.queue_response(404, content=patron_not_found_json)
+
+        assert_raises(PatronNotFoundOnRemote, self.api.release_hold, self.PATRON, self.PIN, self.licensepool)
+
+        print 'Test release hold --> invalid patron ok!'
+
+    def test_62_return_hold_not_found(self):
+        holds_data, holds_json = self.sample_json("patron_holds.json")
+        self.api.queue_response(200, content=holds_json)
+
+        checkin_data, checkin_json = self.sample_json("error_hold_not_found.json")
+        self.api.queue_response(404, content=checkin_json)
+
+        response = self.api.release_hold(self.PATRON, self.PIN, self.licensepool)
+        eq_(response, True,
+            msg="Cannot release hold, response false " + self.licensepool.identifier.identifier + " patron: "
+                + self.PATRON)
+
+        print 'Hold returned: %s' % self.licensepool.identifier.identifier
+
+    def test_63_return_hold(self):
+        print('Return hold test...')
+
+        holds_data, holds_json = self.sample_json("patron_holds.json")
+        self.api.queue_response(200, content=holds_json)
+
+        release_hold_ok_data, release_hold_ok_json = self.sample_json("release_hold_ok.json")
+        self.api.queue_response(200, content=release_hold_ok_json)
+
+        response = self.api.release_hold(self.PATRON, self.PIN, self.licensepool)
+        eq_(response, True,
+            msg="Cannot release hold, response false " + self.licensepool.identifier.identifier + " patron: "
+                + self.PATRON)
+
+        print 'Hold returned: %s' % self.licensepool.identifier.identifier
+
+
+class TestOdiloDiscoveryAPI(OdiloAPITest):
+    def test_1_odilo_recent_circulation_monitor(self):
+        print 'Testing recent library products...'
+
+        monitor = RecentOdiloCollectionMonitor(self._db, self.collection, api_class=MockOdiloAPI)
+        ok_(monitor, 'Monitor null !!')
+        eq_(ExternalIntegration.ODILO, monitor.protocol, 'Wat??')
+
+        records_metadata_data = self.sample_data("records_metadata.json")
+        monitor.api.queue_response(200, content=records_metadata_data)
+
+        availability_data = self.sample_data("record_availability.json")
+        monitor.api.queue_response(200, content=availability_data)
```

Comment:
This would be clearer and take less code if you called this once for every item in `records_metadata_data`.

Reply:
I couldn't agree more, what would I be thinking...?, changed !